apps/builder/app/builder/features/settings-panel/shared.tsx

```diff
@@ -234,75 +239,67 @@ export const useLocalValue = <Type,>(
 
 type LayoutProps = {
   label: ReturnType<typeof Label>;
-  deletable: boolean;
-  onDelete: () => void;
   children: ReactNode;
 };
 
-export const VerticalLayout = ({
-  label,
-  deletable,
-  onDelete,
-  children,
-}: LayoutProps) => (
+export const VerticalLayout = ({ label, children }: LayoutProps) => (
   <Box>
     <Grid
       css={{
-        gridTemplateColumns: deletable ? `1fr max-content` : `1fr`,
+        gridTemplateColumns: `1fr`,
         justifyItems: "start",
       }}
       align="center"
       gap="1"
       justify="between"
     >
       {label}
-      {deletable && <RemovePropButton onClick={onDelete} />}
     </Grid>
     <Box css={{ py: theme.spacing[2] }}>{children}</Box>
   </Box>
 );
 
-export const HorizontalLayout = ({
-  label,
-  deletable,
-  onDelete,
-  children,
-}: LayoutProps) => (
+export const HorizontalLayout = ({ label, children }: LayoutProps) => (
   <Grid
     css={{
-      gridTemplateColumns: deletable
-        ? `${theme.spacing[19]} 1fr max-content`
-        : `${theme.spacing[19]} 1fr`,
+      gridTemplateColumns: `${theme.spacing[19]} 1fr`,
       minHeight: theme.spacing[12],
     }}
     align="center"
     gap="2"
   >
     {label}
     {children}
-    {deletable && <RemovePropButton onClick={onDelete} />}
   </Grid>
 );
 
-export const ResponsiveLayout = ({
-  label,
-  deletable,
-  onDelete,
-  children,
-}: LayoutProps) => {
-  // more than 9 characters in label trigger ellipsis
-  // might not cover all cases though
-  if (label.props.children.length <= 8) {
-    return (
-      <HorizontalLayout label={label} deletable={deletable} onDelete={onDelete}>
-        {children}
-      </HorizontalLayout>
-    );
-  }
+export const ResponsiveLayout = ({ label, children }: LayoutProps) => {
   return (
-    <VerticalLayout label={label} deletable={deletable} onDelete={onDelete}>
-      {children}
-    </VerticalLayout>
+    <Flex
+      align="center"
+      wrap="wrap"
+      css={{
+        columnGap: theme.spacing[5],
+        rowGap: theme.spacing[3],
+        paddingBlock: theme.spacing[2],
+      }}
+    >
+      <Box
+        css={{
+          // wrap label and input when label is more than ~9 characters
```

Comment:
you could use a 9ch

Reply:
if you wanted it to be basde on characters amount

Reply:
I don't. It's hard enough to make it wrap